src/sourcekit-lsp/LanguageClientManager.ts

```diff
@@ -349,7 +349,10 @@ export class LanguageClientManager {
                 .then(async () => {
                     await this.setupLanguageClient(workspaceFolder?.uri);
                 })
-                .catch(reason => {
+                .catch(async reason => {
+                    if (reason.message === "Stopping the server timed out") {
+                        await this.setupLanguageClient(workspaceFolder?.uri);
+                    }
```

Comment:
I don't follow how this guarantees that the "old" client is terminated.  It appears to simply spawn the new instance in the case of a timeout or not.  Am I missing something?

Reply:
I've found that VSCode is not registering that the old server has been killed even though it has. It is a bug in VSCode language client really.